### DISCUSSION TITLE
Feature request: Command Palette

### DISCUSSION THREAD

**Discussion by vladstudio** [mikekazakov/nimble-commander#112] [Ideas]
I know it's a lot to ask :-) but there's a very useful UX pattern I see in various apps, "Command Palette". It's a filtered list of commands, usually invoked with Cmd-K, Cmd-P or Cmd-Shift-P. It's for people like me who prefer to type a couple of letters of a command, instead of having to remember dozens of shortcuts.

https://github.com/mikekazakov/nimble-commander/assets/914320/11fd229f-7c2e-4a3c-8915-5a37650ac273

https://github.com/mikekazakov/nimble-commander/assets/914320/2b16f967-f26a-437c-ac66-6cdc2c5b3a1f

The extreme level is the OS-wide command palette, like Alfred or Raycast:

https://github.com/mikekazakov/nimble-commander/assets/914320/a10601ca-c54b-4ef3-a488-e2fd19876fdd

I imagine the NC command palette could look something like this:

<img width="1120" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/b019730c-5bc4-49e9-a8b6-2292db77a62e">

and its interface would be very similar to Hotkeys window:

https://github.com/mikekazakov/nimble-commander/assets/914320/4e65fdac-a510-45fa-8592-36875fc956de

**Comment by mikekazakov** [comment]
Yes, I agree, that's something I'd also love to have in NC. Just need to implement it :)

**Comment by WWWFWWW** [comment]
Mac OS already offers something like that natively for _menu items_ in every app – under Help/Search, though obviously not for commands without a menu item defined (but it still helps a lot!).

Type any menu item name (or part of a name) in there and it shows up a Spotlight‑like list of the current app's matching menu items, and you can just select the one you want and press enter to execute it.

No idea if that functionality is available as an API though.

But at the very least, you can still configure a system‑wide keyboard shortcut for all apps for "Show Help Menu" in System Preferences, and the shortcut automatically puts the Help/Search entry field into keyboard focus, so just start typing.

Obviously that only works for commands that have a corresponding menu item defined, not commands with just a shortcut and no menu item for them.

**Comment by vladstudio** [comment]
I currently scratch this itch with Raycast, and its wonderful Menu Items extension, tied to Right Option key. Works for every app. But it's still not as native as an in-app command palette.
